bundles/async_remote_services/admin/src/admin.cc

```diff
@@ -21,300 +21,256 @@
+    auto interface = properties->get(celix::rsa::Endpoint::EXPORTS);
 
-    if(interface.empty()) {
+    if (interface.empty()) {
         L_WARN("Adding exported factory but missing exported interfaces");
         return;
     }
 
-    std::unique_lock l(_m);
-    auto factoryIt = _exportedServiceFactories.find(interface);
-
-    if(factoryIt != end(_exportedServiceFactories)) {
-        L_WARN("Adding exported factory but factory already exists");
-        return;
-    }
-
+    std::lock_guard<std::mutex> lock{_m};
     _exportedServiceFactories.emplace(interface, factory);
-    L_WARN("Looping over %i tbce for interface %s", _toBeCreatedExportedEndpoints.size(), interface.c_str());
-
-    for(auto it = _toBeCreatedExportedEndpoints.begin(); it != _toBeCreatedExportedEndpoints.end(); ) {
-        auto interfaceToBeCreated = it->second.get(ENDPOINT_EXPORTS);
-        L_WARN("Checking tbce interface %s", interfaceToBeCreated.c_str());
-
-        if(interfaceToBeCreated.empty() || interfaceToBeCreated != interface) {
-            it++;
-        } else {
-            auto endpointId = it->second.get(ENDPOINT_IDENTIFIER);
-            _exportedServiceInstances.emplace(endpointId, factory->create(it->first, endpointId));
-            it = _toBeCreatedExportedEndpoints.erase(it);
-        }
-    }
+    createExportServices();
 }
 
-void celix::async_rsa::AsyncAdmin::removeExportedServiceFactory(celix::async_rsa::IExportedServiceFactory *, Properties&& properties) {
-    auto interface = properties.get(ENDPOINT_EXPORTS);
+void celix::async_rsa::AsyncAdmin::removeExportedServiceFactory(const std::shared_ptr<celix::async_rsa::IExportedServiceFactory>& /*factory*/, const std::shared_ptr<const celix::Properties>& properties) {
+    auto interface = properties->get(celix::rsa::Endpoint::EXPORTS);
 
     if(interface.empty()) {
         L_WARN("Removing imported factory but missing exported interfaces");
         return;
     }
 
-    std::unique_lock l(_m);
+    std::lock_guard l(_m);
     _exportedServiceFactories.erase(interface);
+
+    //TODO TBD are the removal of the exported services also needed when a ExportedServiceFactory is removed (and maybe the bundle is uninstalled?)
 }
 
-void celix::async_rsa::AsyncAdmin::addService(void *svc, const celix_properties_t *props) {
-    auto *objectClass = celix_properties_get(props, OSGI_FRAMEWORK_OBJECTCLASS, nullptr);
-    auto *remote = celix_properties_get(props, "remote", nullptr);
-    auto endpointId = celix_properties_get(props, ENDPOINT_IDENTIFIER, nullptr);
+void celix::async_rsa::AsyncAdmin::addService(const std::shared_ptr<void>& svc, const std::shared_ptr<const celix::Properties>& props) {
+    assert(props->getAsBool(celix::rsa::REMOTE_SERVICE_PROPERTY_NAME, false)); //"only remote services should be tracker by the service tracker");
```

Comment:
`REMOTE_SERVICE_PROPERTY_NAME` implies a string, if it is a string, `get` is probably a better match than `getAsBool`

Reply:
done